### PR TITLE
bt-9: create input component

### LINKED ISSUE #9: Create Input component
You should create a component which we will reuse throughout the project

Image

https://mui.com/material-ui/react-text-field/

### PR DESCRIPTION
**Pull request by vladjsp**
input

I tried my best. That was so hard to implement all these variants. Very challenging task. I also will return to it for import colors and typography from shared but for now I think it is done.

### CODE REVIEW

File: frontend/src/bundles/common/components/input/input.tsx [hunk 1]
```diff
@@ -1,20 +1,28 @@
...
+
 type Properties<T extends FieldValues> = {
     control: Control<T, null>;
     errors: FieldErrors<T>;
     label: string;
     name: FieldPath<T>;
     placeholder?: string;
-    type?: 'text' | 'email';
+    type?: InputType;
+    disabled?: boolean;
```
Comment (anton-otroshchenko) [inline frontend/src/bundles/common/components/input/input.tsx]: https://github.com/BinaryStudioAcademy/quality-criteria/blob/production/src/javascript.md

B19

**Review by anton-otroshchenko — CHANGES_REQUESTED** [review]

File: frontend/src/bundles/common/components/input/input.tsx [hunk 2]
```diff
@@ -1,20 +1,28 @@
...
     type Control,
     type FieldErrors,
     type FieldPath,
     type FieldValues,
 } from 'react-hook-form';
 
+import searchIcon from '~/assets/img/search.svg';
 import { useFormController } from '~/bundles/common/hooks/hooks.js';
 
+import { getValidClassNames } from '../../helpers/helpers.js';
```
Comment (anton-otroshchenko) [inline frontend/src/bundles/common/components/input/input.tsx]: Use absolute path '~/bundles/common/helpers/helpers.js'

**Review by anton-otroshchenko — CHANGES_REQUESTED** [review]

**Comment by vladjsp** [comment]
I used stylelint-disable in styles one time because that was imposible to overwrite mui helper text color without !important.

**Comment by vladjsp** [comment]
inputs2

**Comment by vladjsp** [comment]
Don't mention top input. It actually has red border while his state == error. Just something happens while I took screenshot.

File: frontend/src/bundles/common/components/input/styles.module.scss [hunk 3]
```diff
@@ -0,0 +1,115 @@
+@import url("../../../../assets/css/styles.scss");
```
Comment (lavandosya) [inline frontend/src/bundles/common/components/input/styles.module.scss]: I think we don't need this import because we already import styles in common file

Comment (vladjsp) [reply frontend/src/bundles/common/components/input/styles.module.scss]: Do you mean index.tsx from the frontend root? Okay, I agree. Should I remove it right now or when I will come back to it next time when Josh or Vova fix all variables with propper names?

Comment (lavandosya) [reply frontend/src/bundles/common/components/input/styles.module.scss]: Maybe it's better to do now but I'm not sure

**Review by lavandosya — APPROVED** [review]

File: frontend/src/bundles/common/components/input/styles.module.scss (hunk 3 above)
Comment (nikita-remeslov) [reply frontend/src/bundles/common/components/input/styles.module.scss]: It is already imported here - frontend/src/index.tsx

File: frontend/src/bundles/common/components/input/input.tsx [hunk 4]
```diff
@@ -24,21 +32,75 @@ const Input = <T extends FieldValues>({
...
+    );
+    const helperTextStyles = getValidClassNames(
+        styles.helperText,
+        hasError && styles.hasError,
+    );
+
     return (
-        <label>
-            <span>{label}</span>
+        <label className={styles.wrapper}>
```
Comment (nikita-remeslov) [inline frontend/src/bundles/common/components/input/input.tsx]: we don't need label, we just need input component

File: frontend/src/bundles/common/components/input/styles.module.scss [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+    padding: 11px 14px 11px 10px;
+}
+
+.inputDisabled {
+    pointer-events: none;
+    background-color: var(--light-background-color);
+}
+
+.hasError {
+    /* stylelint-disable */
```
Comment (nikita-remeslov) [inline frontend/src/bundles/common/components/input/styles.module.scss]: ?

**Review by nikita-remeslov — CHANGES_REQUESTED** [review]

File: frontend/src/bundles/common/components/input/input.tsx (hunk 4 above)
Comment (vladjsp) [reply frontend/src/bundles/common/components/input/input.tsx]: So should I delete from input component all labels and return only TextField. Like this:
```
return (
            <TextField
                {...field}
                type={type}
                placeholder={placeholder}
                error={hasError}
                helperText={(error as string) || ' '}
                className={textFieldRootStyles}
                InputProps={{
                    className: muiInputStyles,
                    disabled: isDisabled,
                    startAdornment: adornment,
                }}
                inputProps={{ className: htmlInputStyles }}
                FormHelperTextProps={{ className: helperTextStyles }}
            />
    );
```
and remobe label from Props of component?

**Review by anton-otroshchenko — APPROVED** [review]

**Review by lavandosya — APPROVED** [review]

**Review by nikita-remeslov — APPROVED** [review]

**Pull request merged**